static/js/components/forms/B2BPurchaseForm.js

```diff
@@ -49,6 +49,13 @@ export const validate = (values: Object) => {
 }
 
 class B2BPurchaseForm extends React.Component<Props> {
+  componentDidMount() {
+    const applyCouponButton = document.getElementById("apply-coupon")
+    if (applyCouponButton) {
+      applyCouponButton.click()
```

Comment:
not good thing to call click in `componentdidmount`, Better to call `applyCoupon` directly here.

Reply:
This way is used to populate the field error in case of an invalid coupon. By direct calling, it wouldn't populate the error on the field.

Reply:
Please pass the reference of setFieldError and setFieldTouched during call.

Reply:
I could see these functions can be used. I might be wrong but I did not see any example of accessing Formik, its values, or its element in the react life cycle method.

Still, in that case, we would have to  and each of 
- get the form from and it's every value from DOM node tree
- then will have to make the object again with all the values in it
- then I will be able to pass that function in that function.

Reply:
I have tried to reference the form as well and tried to call the function but still, this function requires the event which we can't produce without the button click so better way to go that way. If we don't pass an event it stops execution at `event.preventDefault` with an error in the console.

Reply:
@umarmughal824  Can you give a try to this here ? 

```
const formikRef = this.formikRef.current
if (formikRef.state.values.coupon) {
      this.applyCoupon(
        formikRef.state.values,
        formikRef.setFieldError,
        formikRef.setFieldTouched,
        null
      )
  }
```
Below, try to give reference to form. 
```
 <Formik
     ref={this.formikRef}
     onSubmit={onSubmit}
     ...
```

Reply:
@asadiqbal08 please read the last message carefully. I have tried it already.

Reply:
then I believe it should work.